testcases/kernel/syscalls/madvise/madvise07.c

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2016-2017 Richard Palethorpe <richiejp@f-m.fm>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * Check that accessing memory marked with MADV_HWPOISON results in SIGBUS.
+ *
+ * Test flow: map memory, 
+ *	      create child process, 
+ *	      mark memory with MADV_HWPOISON inside child process,
+ *	      access memory,
+ *	      if SIGBUS is delivered to child the test passes else it fails
+ */
+
+#include "tst_test.h"
+#include <fcntl.h>
+#include <unistd.h>
+#include <signal.h>
+
+#ifdef MADV_HWPOISON
+
+struct test_case {
+	int fd;
+	char *fname;
+	char *fcontent;
+	void *fmem;
+	struct stat stat;
+};
+static struct test_case cases[1];
+
+static void setup(void)
+{
+	cases[0].fd = 0;
+	cases[0].fcontent = "abcdefghijklmnopqrstuvwxyz12345\n";
+	cases[0].fname = "mmf0";
+	cases[0].fmem = NULL;
+}
+
+static void cleanup(void)
+{
+	unsigned int i;
+	struct test_case c;
+
```

Comment:
Structures are passed around as a pointers in C, since otherwise each assignment copies the content.